thoth/metrics_exporter/jobs.py

```diff
@@ -55,257 +46,427 @@
 
-            for j_status, j_counts in jobs_status_evaluated.items():
-                metrics.jobs_status.labels(label_selector, j_status, namespace).set(j_counts)
+            for label in self._JOBS_LABELS + ["operator=graph-sync", "operator=workload"]:
+                _LOGGER.info("Evaluating ConfigMaps(label_selector=%r) metrics for namespace: %r", label, namespace)
+                config_maps_items = self._OPENSHIFT.get_configmaps(namespace=namespace, label_selector=label)
+                number_configmaps = self.count_configmaps(config_maps_items)
+                metrics.config_maps_number.labels(namespace, label).set(number_configmaps)
+                _LOGGER.debug(
+                    "thoth_config_maps_number=%r, in namespace=%r for label=%r", number_configmaps, namespace, label
+                )
+
+
+class CephMetrics:
+    """Class to evaluate Metrics for Ceph."""
+
+    _MONITORED_STORES = (
+        AdvisersResultsStore(),
+        AnalysisResultsStore(),
+        InspectionResultsStore(),
+        ProvenanceResultsStore(),
+        PackageAnalysisResultsStore(),
+        SolverResultsStore(),
+        DependencyMonkeyReportsStore(),
+    )
+
+    def get_ceph_results_per_type(self):
+        """Get the total number of results in Ceph per type."""
+        for store in self._MONITORED_STORES:
+            _LOGGER.info("Check Ceph content for %s", store.RESULT_TYPE)
+            if not store.is_connected():
+                store.connect()
+            all_document_ids = store.get_document_listing()
+            list_ids = [str(cid) for cid in all_document_ids]
+            metrics.ceph_results_number.labels(store.RESULT_TYPE).set(len(list_ids))
+            _LOGGER.debug("ceph_results_number for %s =%d", store.RESULT_TYPE, len(list_ids))
+
+    def get_ceph_connection_error_status(self):
+        """Check connection to Ceph instance."""
+        inspections = InspectionResultsStore()
+        try:
+            inspections.connect()
+        except Exception as excptn:
+            metrics.ceph_connection_error_status.set(1)
+            _LOGGER.exception(excptn)
+        else:
+            metrics.ceph_connection_error_status.set(0)
+
+
+class DBMetrics:
+    """Class to evaluate Metrics for Thoth Database."""
+
+    def get_graphdb_connection_error_status(self):
+        """Raise a flag if there is an error connecting to database."""
+        graph_db = GraphDatabase()
+        try:
+            graph_db.connect()
+        except Exception as excptn:
+            metrics.graphdb_connection_error_status.set(1)
+            _LOGGER.exception(excptn)
+        else:
+            metrics.graphdb_connection_error_status.set(0)
+
+    def get_tot_records_count(self):
+        """Get the total number of Records in Thoth Knowledge Graph."""
+        graph_db = GraphDatabase()
+        graph_db.connect()
+
+        main_models_record_count = sum([r for r in graph_db.get_number_main_tables_records().values()])
```

Comment:
Why not just sum of values?


```python
        main_models_record_count = sum(graph_db.get_number_main_tables_records().values())
```

In other parts as well

Reply:
Correct, thanks!